ext/storage_sources/azure_store/test/test_azure_connection.cpp

```diff
@@ -30,5 +30,176 @@
+// generated test prefix: "azuretest/unit/2023-31-01-16-34-10/623843294--".
+static int
+randomize_test_prefix()
+{
+    char time_str[100];
+    std::time_t t = std::time(nullptr);
+
+    REQUIRE(std::strftime(time_str, sizeof(time_str), "%F-%H-%M-%S", std::localtime(&t)) != 0);
+
+    obj_prefix += time_str;
+
+    // Create a random device and use it to generate a random seed to initialize the generator.
+    std::random_device my_random_device;
+    unsigned seed = my_random_device();
+    std::default_random_engine my_random_engine(seed);
+
+    obj_prefix += '/' + std::to_string(my_random_engine());
+    obj_prefix += "--";
+
+    return 0;
+}
+
+TEST_CASE("Testing Azure Connection Class", "azure-connection")
+{
+    auto azure_client = Azure::Storage::Blobs::BlobContainerClient::CreateFromConnectionString(
+      std::getenv("AZURE_STORAGE_CONNECTION_STRING"), "myblobcontainer1");
+    bool exists = false;
+
+    REQUIRE(randomize_test_prefix() == 0);
+
+    azure_connection conn = azure_connection("myblobcontainer1", obj_prefix);
+    azure_connection conn_bad = azure_connection("myblobcontainer1", "bad_prefix_");
+
+    // Payloads for blobs uploaded to container.
+    const std::string payload = "payload";
+    const std::string payload_2 = "Testing offset and substring";
+
+    void *buffer = calloc(1024, sizeof(char));
+    std::vector<std::string> objects;
+    std::vector<std::pair<std::string, std::string>> blob_objects;
+
+    const std::string object_name = "test_object";
+    const std::string object_name_2 = "test_object_2";
+    const std::string non_exist_object_key = "test_non_exist";
+    blob_objects.push_back(std::make_pair(object_name, payload));
+    blob_objects.push_back(std::make_pair(object_name_2, payload_2));
+
+    for (auto pair : blob_objects) {
+        auto blob_client = azure_client.GetBlockBlobClient(obj_prefix + pair.first);
+        blob_client.UploadFrom(create_file(pair.first, pair.second));
+    }
+
+    SECTION("Check Azure connection constructor.", "[azure-connection]")
+    {
+        REQUIRE_THROWS_WITH(azure_connection("Bad_bucket", ""), "Bad_bucket : No such bucket.");
+    }
+
+    SECTION("Check object exists in Azure.", "[azure-connection]")
+    {
+        // Object exists so there should be 1 object.
+        REQUIRE(conn.object_exists(object_name, exists) == 0);
+        REQUIRE(exists == true);
+
+        // Object does not exist so there should be 0 objects.
+        REQUIRE(conn.object_exists(non_exist_object_key, exists) == 0);
+        REQUIRE(exists == false);
+    }
+
+    SECTION("List Azure objects under the test bucket.", "[azure-connection]")
+    {
+        // No matching objects. Object size should be 0.
+        objects.clear();
+        REQUIRE(conn.list_objects(obj_prefix + non_exist_object_key, objects, false) == 0);
+        REQUIRE(objects.size() == 0);
+
+        // No matching objects with listSingle. Object size should be 0.
```

Comment:
Don't think we have `listSingle` variable, so let's use more "list single functionality"

Reply:
agreed